domains/client/domain-operator/src/domain_bundle_proposer.rs

```diff
@@ -58,12 +62,33 @@ impl<Block: BlockT, CBlock: BlockT> PreviousBundledTx<Block, CBlock> {
     }
 }
 
+pub fn calculate_max_bundle_weight_and_size(
+    domain_block_limit: DomainBlockLimit,
+    consensus_slot_probability: (u64, u64),
+    bundle_slot_probability: (u64, u64),
+) -> (Weight, u32) {
+    // (n1 / d1) / (n2 / d2) is equal to (n1 * d2) / (d1 * n2)
+    // This represents: bundle_slot_probability/SLOT_PROBABILITY
+    let expected_bundles_per_block = (bundle_slot_probability.0 * consensus_slot_probability.1)
+        / (bundle_slot_probability.1 * consensus_slot_probability.0);
+
+    // This represents: Ceil[2*Sqrt[bundle_slot_probability/SLOT_PROBABILITY]])
+    let std_of_expected_bundles_per_block = (2 * expected_bundles_per_block.integer_sqrt()) + 1;
+
+    // max_bundle_weight = TargetDomainBlockWeight/(bundle_slot_probability/SLOT_PROBABILITY+ Ceil[2*Sqrt[ bundle_slot_probability/SLOT_PROBABILITY]])
+    let max_bundle_weight = domain_block_limit.max_block_weight
+        / (expected_bundles_per_block + std_of_expected_bundles_per_block);
+
+    (max_bundle_weight, domain_block_limit.max_block_size)
```

Comment:
@NingLin-P do you recall if we discussed anything about limiting bundle size too based on expected number of bundles?

Reply:
Yes, I believe we also need to limit the bundle size here.

Reply:
@ParthDesai let's do it the same way as weight then:
```suggestion
    let max_bundle_size = domain_block_limit.max_block_size
        / (expected_bundles_per_block + std_of_expected_bundles_per_block);
    (max_bundle_weight, max_bundle_size)
```